template.json

```diff
@@ -319,7 +321,8 @@
     },
     "talent": {
       "templates": ["rollModifiers"],
-      "rank": 1,
+      "type": "general",
+      "rank": "",
```

Comment:
I'd prefer if we keep the `rank` default as `1` and don't start mixing types if we can avoid it.

Reply:
As in you'd rather not have a type property in Talents, or you'd prefer renaming "type" to something else to avoid internal confusion with parent "type"?

Reply:
No, my comment was regarding the change to rank. I prefer if we keep `"rank": 1` as before.

Your comment about `type` is valid too, but I don't have a strong opinion there.

Reply:
> No, my comment was regarding the change to rank. I prefer if we keep `"rank": 1` as before.
> 
> Your comment about `type` is valid too, but I don't have a strong opinion there.

The reason I set rank to `blank` here was because of the way I set up talents in the compendia, where rank should technically be a feature on the character sheet and not the item. This is because I don't split Talents by rank, instead all ranks are in a single Talent item. I have manually removed Rank 1 from every talent to accommodate this.

I think any solution here will be opinionated one way or another.

Reply:
Sure, I see your point with `rank` not belonging to the talent itself but rather to the relation between actor and talent. I don't see how that calls for setting its default value to `""`, though. It's still good practice to not mix types.

The situation is similar to classes in DnD 5e, where the `levels` property does not technically belong to the class item itself. If you look at the implementation in that system you will see that the default value used in both `template.json` and the compendium packs is the integer `1`.

In the end though, I don't think this matters much! Javascript is losely typed and we don't define a `data-dtype` for the rank input fields anyway, so it's all going to end up as string eventually anyway. I've taken my stand, and now we can move on! 😄